src/components/entity/ha-chart-base.js

```diff
@@ -537,71 +537,22 @@ class HaChartBase extends mixinBehaviors(
   static getColorGenerator(staticColors, startIndex) {
     // Known colors for static data,
     // should add for very common state string manually.
-    // Palette modified from http://google.github.io/palette.js/ mpn65, Apache 2.0
+    // Palette modified from https://www.materialui.co/colors
     const palette = [
-      "ff0029",
-      "66a61e",
-      "377eb8",
-      "984ea3",
-      "00d2d5",
-      "ff7f00",
-      "af8d00",
-      "7f80cd",
-      "b3e900",
-      "c42e60",
-      "a65628",
-      "f781bf",
-      "8dd3c7",
-      "bebada",
-      "fb8072",
-      "80b1d3",
-      "fdb462",
-      "fccde5",
-      "bc80bd",
-      "ffed6f",
-      "c4eaff",
-      "cf8c00",
-      "1b9e77",
-      "d95f02",
-      "e7298a",
-      "e6ab02",
-      "a6761d",
-      "0097ff",
-      "00d067",
-      "f43600",
-      "4ba93b",
-      "5779bb",
-      "927acc",
-      "97ee3f",
-      "bf3947",
-      "9f5b00",
-      "f48758",
-      "8caed6",
-      "f2b94f",
-      "eff26e",
-      "e43872",
-      "d9b100",
-      "9d7a00",
-      "698cff",
-      "d9d9d9",
-      "00d27e",
-      "d06800",
-      "009f82",
-      "c49200",
-      "cbe8ff",
-      "fecddf",
-      "c27eb6",
-      "8cd2ce",
-      "c4b8d9",
-      "f883b0",
-      "a49100",
-      "f48800",
-      "27d0df",
-      "a04a9b",
+      "FF5722",
+      "4CAF50",
+      "FF9800",
+      "FFEE58",
+      "009688",
+      "00BCD4",
+      "3F51B5",
+      "673AB7",
+      "E91E63",
     ];
     function getColorIndex(idx) {
       // Reuse the color if index too large.
-      return Color("#" + palette[idx % palette.length]);
+      const styles = getComputedStyle(document.body);
+      return Color(styles.getPropertyValue(`--graph-color-${idx % palette.length}`) || "#" + palette[idx % palette.length]);
```

Comment:
This means we will never reuse custom colors, so if a chart has 50 different values it will need 50 custom colors and otherwise fallback to the default palette.

Reply:
It does mean that you can't change the length of the palette from a theme, but I don't see a problem other than that. From what I understand, it's looking for the variable, and falling back to the palette if it doesn't exist.